101-azure-spring-cloud/azuredeploy.json

```diff
@@ -0,0 +1,183 @@
+                    ],
+                    "resources": [
+                        {
+                            "type": "deployments",
+                            "apiVersion": "2020-07-01",
+                            "name": "default",
+                            "dependsOn": [
+                                "[parameters('app1')]"
+                            ],
+                            "properties": {
+                                "source": {
+                                    "relativePath": "<default>",
+                                    "type": "Jar"
+                                }
+                            }
+                        }
+                    ]
+                },
+                {
+                    "type": "apps",
+                    "apiVersion": "2020-07-01",
+                    "name": "[parameters('app2')]",
+                    "properties": {
+                        "public": true
+                    },
+                    "dependsOn": [
+                        "[parameters('serviceName')]"
+                    ],
+                    "resources": [
+                        {
+                            "type": "deployments",
+                            "apiVersion": "2020-07-01",
+                            "name": "default",
+                            "dependsOn": [
+                                "[parameters('app2')]"
+                            ],
+                            "properties": {
+                                "source": {
+                                    "relativePath": "<default>",
+                                    "type": "Jar"
+                                }
+                            }
+                        }
+                    ]
+                },
+                {
+                    "type": "apps",
+                    "apiVersion": "2020-07-01",
+                    "name": "[parameters('app3')]",
+                    "properties": {
+                        "public": false
+                    },
+                    "dependsOn": [
+                        "[parameters('serviceName')]"
+                    ],
+                    "resources": [
+                        {
+                            "type": "deployments",
+                            "apiVersion": "2020-07-01",
+                            "name": "default",
+                            "dependsOn": [
+                                "[parameters('app3')]"
+                            ],
+                            "properties": {
+                                "source": {
+                                    "relativePath": "<default>",
+                                    "type": "Jar"
+                                }
+                            }
+                        }
+                    ]
+                }
+            ]
+        },
+        {
+            "type": "Microsoft.Resources/deployments",
```

Comment:
add some explanation to the readme.md to explain why we're deploying these twice.

Reply:
Will wait for review/input from the product team, unless you know why it's deployed twice

Reply:
"name": "setActiveDeployment",
As the name indicates, the second time to deploy is to set the "deployments" as "active" for the corresponding apps. Before doing this all resource are provisioned but traffic to apps is not routed to the running "deployments".